lightning/src/ln/channelmanager.rs

```diff
@@ -2946,17 +2951,12 @@ where
 			}
 		};
 		if let Some(update) = update_opt {
-			// Try to send the `BroadcastChannelUpdate` to the peer we just force-closed on, but if
-			// not try to broadcast it via whatever peer we have.
-			let per_peer_state = self.per_peer_state.read().unwrap();
-			let a_peer_state_opt = per_peer_state.get(peer_node_id)
-				.ok_or(per_peer_state.values().next());
-			if let Ok(a_peer_state_mutex) = a_peer_state_opt {
-				let mut a_peer_state = a_peer_state_mutex.lock().unwrap();
-				a_peer_state.pending_msg_events.push(events::MessageSendEvent::BroadcastChannelUpdate {
-					msg: update
-				});
-			}
+			// If we have some Channel Update to broadcast, we cache it and broadcast it later.
+			let mut pending_broadcast_messages = self.pending_broadcast_messages.lock().unwrap();
+			pending_broadcast_messages.push(events::MessageSendEvent::BroadcastChannelUpdate {
+				msg: update
+			});
+			log_trace!(self.logger, "Caching the channel_updates of force-closed channel. We will them broadcast later.");
```

Comment:
I get your point.
But I think the phrase, "Broadcasting channel_update [once we have connected peers]" might be a bit confusing too
This phrasing's kinda implying that we are not connected to any peers right now.

I'm sure there might be a better way to phrase it, but I remembered what Wpaulino mentioned earlier in a [comment](https://github.com/lightningdevkit/rust-lightning/pull/2731#discussion_r1459618170):

> I don't think this needs to be logged at all...
> 

Considering that, I'm starting to wonder if logging this message is necessary at all, especially since the behavior is pretty clear from the code snippet:

```rust
	pending_broadcast_messages.push(events::MessageSendEvent::BroadcastChannelUpdate {
				msg: update
			});

```

Let me know what you think!

I'm leaning towards temporarily removing the logging. It seems like it could clean things up a bit, but I'm open to your thoughts on it!